ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -74,10 +75,49 @@ public class BlockTransactionSelector {
   private final Wei minTransactionGasPrice;
   private final Double minBlockOccupancyRatio;
 
+  public static class TransactionValidationResult {
+    private final Transaction transaction;
+    private final ValidationResult<TransactionInvalidReason> validationResult;
+
+    public TransactionValidationResult(
+        final Transaction transaction,
+        final ValidationResult<TransactionInvalidReason> validationResult) {
+      this.transaction = transaction;
+      this.validationResult = validationResult;
+    }
+
+    public Transaction getTransaction() {
+      return transaction;
+    }
+
+    public ValidationResult<TransactionInvalidReason> getValidationResult() {
+      return validationResult;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      TransactionValidationResult that = (TransactionValidationResult) o;
```

Comment:
if we can assume that a transaction can only appear once in this list, then could be enough to check the tx hash for equality

Reply:
I am afraid we can't make that assumption. This is being used by 
```java 
// AbstractBlockCreator class 
BlockCreationResult createBlock(
      final Optional<List<Transaction>> maybeTransactions,
      final Optional<List<BlockHeader>> maybeOmmers,
      final Optional<Bytes32> maybePrevRandao,
      final long timestamp,
      boolean rewardCoinbase)
```
`maybeTransactions` can have repeated transactions.
So, I think it would be safer to keep it as it is.